OsuPlayer.Data/OsuPlayer/Classes/ShuffleAlgorithm.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Text.RegularExpressions;
+
+namespace OsuPlayer.Data.OsuPlayer.Classes;
+
+public class ShuffleAlgorithm
+{
+    public Type Type { get; }
+    private readonly string _formattedName;
```

Comment:
Maybe Instead of using a formated name, the ShuffleImplementation Interface provides an Shuffle Name, so we can use that instead of using a RGX, which also gives more ability with custom names.

Reply:
Yes, but we don't have access to the class itself, only to it's runtime type. That's why there isn't a good solution to access the properties and therefore a name.

Reply:
But it defininitly looks like we can still resolve gain access to the class itself to access that propert. Because if you can access and use the formated Regex, we can also use a custom name

Reply:
The formatted Regex is based on the type name, which is the same as the class name. To make it easier, I would go for a custom attribute like the DefaultImpl one, which provides a custom name and a small description so we can access this at runtime and don't need to do heavy reflection stuff to access properties.

Reply:
Ok, I see the issue noe, wasnt really awake haha.

Anyhow, instead of using a Attribute, IIRC we can change the Type to something like an Interface instead of Type. This will already give us the option to acces the right members.

Or if you want to continue using Type as Membertype. we can just call Type.ToString() and ever ShuffleImpl class overrides ToString.

This sounds like the most simple solution without any overhead without unneeded reflection nor attributes.

Reply:
I could built it in a way which does not require loading the shuffle implementation as a service and instead only provide it through the shuffle service provider. We could then use the interface we already have (IShuffleImpl) to store that information. Maybe this would be better yes.

Reply:
Thats a solution I can agree with. We dont need everything as a service if we can add it usefully to an existing one

Reply:
I changed it, but now we need to explicitly hold an instance of each shuffle implementation in the provider, because if we don't, we can not access the properties of the interface, as they are instance bound. Now we have a memory overhead or do you have a better idea?

Reply:
What do you mean memory overhead? It's not like this will suck a lot of memory ao I wouldn't even consider this bad. if you want elaborate this more. Also its fine to have at least one instance of shuffle impl. which makes sense, because people qant to use shuffle and we always need one and this is fine.

Reply:
No it does not use much memory, but we will need _every_ shuffle implementation instance in memory to access the properties for the ui e.g. name, description.
If we would have like 20 different shuffle implementations, we would need to have all 20 as an instance in memory.
Before that, we could load them "lazily" when needed/when selected.
But yes, 3 shuffle implementations which only one needs "big amount" of memory (at 20k maps it's around 80KB) doesn't need that much, but this is to keep in mind.